### PR TITLE
feat!: move operations to nested Auth and User structs

### PR DESCRIPTION
**Pull request by ctran88**
## What's New?

- moves auth operations under a nested `Auth` struct
- moves user operations under a nested `User` struct
- keeps the `set_server_url` func to maintain tests until they are moved to the e2e suite
- removes the `get_app` func since it doesn't really serve much purpose except for a test of the sdk without inducing any side effects

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have manually tested my code thoroughly
- [ ] I have added/updated inline documentation for public facing interfaces if relevant
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing integration and unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -62,5 +62,7 @@ pub mod models;
 #[rustfmt::skip]
 pub mod openapi;
 
+pub mod auth;
```
Comment (ctran88) [inline src/lib.rs:65]: i think this would also expose the auth and user structs to the developer

File: src/passage_flex.rs [hunk 2]
```diff
@@ -43,373 +37,27 @@ impl PassageFlex {
...
-                    updated_at: response.user.updated_at,
-                    user_metadata: response.user.user_metadata,
-                    webauthn: response.user.webauthn,
-                    webauthn_devices: response.user.webauthn_devices,
-                    webauthn_types: response.user.webauthn_types,
-                })
-            })
-            .map_err(Into::into)
+        self.user.configuration.base_path = format!("{}/v1/apps/{}", server_url, self.app_id);
+        self.auth.configuration.base_path = format!("{}/v1/apps/{}", server_url, self.app_id);
```
Comment (ctran88) [inline src/passage_flex.rs:60]: necessary to mock the server calls in the unit tests. this can be removed once the tests are moved to the e2e suite

**Comment by bertrmz** [comment]
The repo's [README](https://github.com/passageidentity/passage-flex-rust/blob/f402b262a196920b5f383142789421002914bc9a/README.md) will also need updates to reference the nested structure.

**Comment by ctran88** [comment]
> The repo's [README](https://github.com/passageidentity/passage-flex-rust/blob/f402b262a196920b5f383142789421002914bc9a/README.md) will also need updates to reference the nested structure.

updated in [a1540af](https://github.com/passageidentity/passage-flex-rust/pull/30/commits/a1540af4ccfd4ba260c0d85caf9685cc0de9be30)

File: README.md [hunk 3]
```diff
@@ -158,13 +144,13 @@ let passage_flex = PassageFlex::new(
 let external_id = your_user.id;
 
 // get user info
-let user_info = passage_flex.get_user(external_id).await.unwrap();
+let user_info = passage_flex.user.get_user(external_id).await.unwrap();
```
Comment (bertrmz) [inline README.md:147]:
```suggestion
let user_info = passage_flex.user.get(external_id).await.unwrap();
```

**Comment by bertrmz** [comment]
There's a usage example in lib.rs that will need to be updated: https://github.com/passageidentity/passage-flex-rust/blob/c8f08402acd720d30884987b7223c583489ad4c4/src/lib.rs#L14

File: README.md (hunk 3 above)
Comment (ctran88) [reply README.md]: ah damn it [efd394e](https://github.com/passageidentity/passage-flex-rust/pull/30/commits/efd394e71e4c42a812767c313a0eedc912dc68d3)

**Comment by ctran88** [comment]
> There's a usage example in lib.rs that will need to be updated:

[80ef1f4](https://github.com/passageidentity/passage-flex-rust/pull/30/commits/80ef1f4f6dfd58398bed5a24d768221b4f5edffc)

also removed the `GetAppError` in [f22a04e](https://github.com/passageidentity/passage-flex-rust/pull/30/commits/f22a04ed9e892fbb82313de365b6c06218d26190)

File: src/auth.rs [hunk 4]
```diff
@@ -0,0 +1,141 @@
+use crate::openapi::apis::configuration::Configuration;
+use crate::openapi::apis::{authenticate_api, transactions_api};
+use crate::Error;
+
+pub struct Auth {
+    pub(crate) configuration: Configuration,
```
Comment (bertrmz) [inline src/auth.rs:6]: This looks like it'd make the configuration of the struct public within the crate.
Where are the internal crate modules accessing this field?

File: src/error.rs [hunk 5]
```diff
@@ -16,17 +14,6 @@ fn convert_error<Src>(error: crate::openapi::apis::Error<Src>, map_fn: fn(Src) -
     }
 }
 
-impl From<crate::openapi::apis::Error<apps_api::GetAppError>> for Error {
-    fn from(e: crate::openapi::apis::Error<apps_api::GetAppError>) -> Self {
```
Comment (bertrmz) [inline src/error.rs:20]: Will `GetApp` need to be removed from the codegen output as well?
Would that be a separate PR?

File: src/lib.rs [hunk 6]
```diff
@@ -11,8 +11,8 @@
 //!     std::env::var("PASSAGE_API_KEY").unwrap(),
 //! );
 //!
-//! let app_info = passage_flex.get_app().await.unwrap();
-//! println!("{}", app_info.auth_origin);
+//! let passage_user = passage_flex.user.get(external_id).await.unwrap();
+//! println!("{:?}", passage_user.webauthn_devices);
```
Comment (bertrmz) [inline src/lib.rs:15]: It might be simpler to remove this part of the usage example. The previous method was clear on its own, but most flex methods need context (e.g., what’s an `external_id`?).

File: src/auth.rs (hunk 4 above)
Comment (ctran88) [reply src/auth.rs]: ah sorry this was leftover from when i was trying something earlier [9895501](https://github.com/passageidentity/passage-flex-rust/pull/30/commits/98955014feda994b816cbe8c2d51e99a7fc987f9)

File: src/lib.rs (hunk 6 above)
Comment (ctran88) [reply src/lib.rs]: removed in [b7160f9](https://github.com/passageidentity/passage-flex-rust/pull/30/commits/b7160f9d857f65d0fbf587901f04b07c68ba74a5)

File: src/error.rs (hunk 5 above)
Comment (ctran88) [reply src/error.rs:20]: i was planning on removing the flex sdk annotation from the schema in order to remove it in a codegen pr

File: src/auth.rs [hunk 7]
```diff
@@ -0,0 +1,141 @@
+use crate::openapi::apis::configuration::Configuration;
+use crate::openapi::apis::{authenticate_api, transactions_api};
+use crate::Error;
+
+pub struct Auth {
+    pub configuration: Configuration,
```
Comment (bertrmz) [inline src/auth.rs:6]: Does this field need to be public? I’d expect it to be encapsulated in the struct after initialization, but let me know if I’m missing a use case.

File: src/user.rs [hunk 8]
```diff
@@ -0,0 +1,180 @@
+use crate::models::PassageUser;
+use crate::openapi::apis::configuration::Configuration;
+use crate::openapi::apis::{user_devices_api, users_api};
+use crate::Error;
+
+pub struct User {
+    pub(crate) configuration: Configuration,
```
Comment (bertrmz) [inline src/user.rs:7]: This field's visibility will need to be updated to match the similar field in the `Auth` struct.

File: src/auth.rs (hunk 7 above)
Comment (ctran88) [reply src/auth.rs:6]: the parent `set_server_url` method is used in testing and needs to set each nested class's base path for the tests to mock correctly

https://github.com/passageidentity/passage-flex-rust/blob/b7160f9d857f65d0fbf587901f04b07c68ba74a5/src/passage_flex.rs#L58-L61

Comment (ctran88) [reply src/auth.rs:6]: i started messing with passing a reference, but it got kinda messy with the lifecycle annotations so decided to just make them pub

File: src/user.rs (hunk 8 above)
Comment (ctran88) [reply src/user.rs]: [ac297a2](https://github.com/passageidentity/passage-flex-rust/pull/30/commits/ac297a257a3e1abdf3c901aa08c8528d23c478fd)

File: src/auth.rs (hunk 4 above)
Comment (ctran88) [reply src/auth.rs:6]: to circle back, the `(crate)` modifier was necessary to keep compatibility with the test mocks via the `set_server_url` method in the base class. dropped the above commit in a rebase to keep the modifier

**Review by bertrmz — APPROVED** [review]
